src/ruleChecker.ts

```diff
@@ -126,6 +127,18 @@ export default class RuleChecker {
       hash.update(findingEvent.hash);
       hash.update(checkInstance.ruleId);
 
+      const uniqueEvents = new Set<number>();
+      const relatedEvents: Array<Event> = [];
+      [cloneEvent(findingEvent)]
+        .concat((additionalEvents || []).map((event) => cloneEvent(event)))
+        .forEach((event) => {
```

Comment:
I find `forEach` harder to read than `for ... of` loops, and many style guides discourage them.